### PR TITLE
#1212: Add PasswordInput component (closes #1212)

### LINKED ISSUE #1212: Add PasswordInput component
[Project card](https://github.com/buildo/react-components/projects/1?fullscreen=true#card-8552800)

## requirements
{describe the new feature}

## specs
it should be based on `Input`

## misc
{optional: other useful info}

### PR DESCRIPTION
**Pull request by giogonzo**
Closes #1212

## Test Plan

> These results can be screenshots, query results, or even just pointers to unit tests that successfully passed.

- [ ] Google Chrome
- [ ] Internet Explorer

> An example of partial test is a field displaying 4 possible values. If 3 values are tested, with screenshots, and 1 is not, then it should be mentioned here.

### CODE REVIEW

**Comment by giogonzo** [comment]
@FrancescoCioria fixed tests

File: src/PasswordInput/Examples.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
+### Examples
+
+```js
+initialState = { value: '' };
+
+const onChange = value => setState({ value });
+
+<form className='ui form'>
```
Comment (FrancescoCioria) [inline src/PasswordInput/Examples.md]: this is a left-over from when we used semantic's css, you can remove the `className` and, if you want, the `form` as well

File: src/PasswordInput/PasswordInput.tsx [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+};
+
+export namespace PasswordInput {
+  export type Props = PasswordInputRequiredProps & Partial<PasswordInputDefaultProps>;
+}
+type PasswordInputDefaultedProps = PasswordInputRequiredProps & PasswordInputDefaultProps;
+
+export const Props = {
+  ...InputPropsT,
+  hideText: t.maybe(t.String),
```
Comment (FrancescoCioria) [inline src/PasswordInput/PasswordInput.tsx]: can you add a description for these two new props?

File: src/PasswordInput/PasswordInput.tsx [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+    this.setState(({ show }) => ({ show: !show }));
+  }
+
+  render() {
+    const props = this.props as PasswordInputDefaultedProps;
+    const type = this.state.show ? 'text' : 'password';
+    const text = this.state.show ? props.hideText : props.showText;
+    const toggleProps = {
+      onClick: this.onToggleClick,
+      className: 'toggle',
```
Comment (FrancescoCioria) [inline src/PasswordInput/PasswordInput.tsx]: this `className` is named as one of our components => `Toggle`'s CSS properties may be applied to this.
CSS modules may be a long-term solution for these problems, but for now I'd simply rename it

**Comment by giogonzo** [comment]
@FrancescoCioria done!

**Pull request merged**